test/utils/dartesr_test.dart

```diff
@@ -1,39 +1,78 @@
 
 void main() {
   group('dartesr', () {
     test('decode esr', () async {
-      final client =
-          EOSClient('https://telos.caleos.io', 'v1', privateKeys: []);
+      var esr = 'esr://gmNcs7jsE9uOP6rL3rrcvpMWUmN27LCdleD836_eTzFz-vCSjQEMXhmEFohe6ry3yuguIyNEiIEJSgvCBA58nnUl1dgwlAEoAAA';
 
-      final esr = await EosioSigningRequest.factory(
-        client,
-        'esr:gmNgYmAoCOJqXqlwloGBIVzX5uxZRkYGCGCC0oowAYeHIR9XFWsWFwhCRFiCXV1dgoEMIBcA',
-        'sevenflash42',
-      );
+      var request = SeedsESR(uri: esr);
 
-      Map<String, dynamic> data = Map<String, dynamic>.from(esr.action.data);
+      await request.resolve(account: "bukabukabuka");
 
-      expect(esr.action.account, 'token.seeds');
-      expect(esr.action.name, 'transfer');
-      expect(data['from'], 'sevenflash42');
+      print("1 actions[0]: "+request.firstAction().toJson().toString());
+
+      var action = request.firstAction();
+      var data = Map<String, dynamic>.from(action.data);
+
+      expect(request.firstAction().account, 'eosio');
+      expect(request.firstAction().name, 'voteproducer');
+      expect(data['voter'], 'bukabukabuka');
+      expect(data['proxy'], 'eosnationftw');
+      expect(data['producers'], []);
+    });
+
+    test('decode esr 2', () async {
+      var esr = 'esr://gmNgYmAoCOJqXqlwloGBIVzX5uxZRkYGCGCC0oowAYeHIR9XFWsWFwhCRFiCXV1dgoEMIBcA';
+
+      var request = SeedsESR(uri: esr);
+
+      await request.resolve(account: "bukabukabuka");
+
+      print("2 actions[0]: "+request.firstAction().toJson().toString());
+
+      var action = request.firstAction();
+      var data = Map<String, dynamic>.from(action.data);
+
+      expect(request.firstAction().account, 'token.seeds');
+      expect(request.firstAction().name, 'transfer');
+      expect(data['from'], 'bukabukabuka');
       expect(data['to'], 'igorberlenko');
       expect(data['quantity'], '7.0000 SEEDS');
       expect(data['memo'], '');
+
+    });    test('decode esr 3', () async {
+      var esr = 'esr:gmN0S9_Eeqy57zv_9xn9eU3hL_bxCbUs-jptJqsXY3-JtawgU09qb_yjZdUau3YzAAEjQ0EQV_NKhbNAdriuzdmzjIwMEMAEpTVgAgeWdbM9zBDoDTgM4bMEu7q6BDMwsJekFpdk5qUDzWIAAA';
+
+      var request = SeedsESR(uri: esr);
+
+      await request.resolve(account: "illumination");
+
+      //print("actions: "+actions.toString());
+      print("3 actions[0]: "+request.firstAction().toJson().toString());
+
+      var action = request.firstAction();
+      var data = Map<String, dynamic>.from(action.data);
+
+      expect(request.firstAction().account, 'token.seeds');
+      expect(request.firstAction().name, 'transfer');
+      expect(data['from'], 'illumination');
+      expect(data['to'], 'localscaling');
+      expect(data['quantity'], '5.0000 SEEDS');
+      expect(data['memo'], 'testing');
     });
+    
     test('login signing', () async {
-      final client =
-          EOSClient('https://telos.caleos.io', 'v1', privateKeys: []);
+      // final client =
```

Comment:
I know its test code, still dont like dead code

Reply:
made this test work instead